plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMigrateProcedure.java

```diff
@@ -302,24 +305,76 @@ private String getColumnComment(String tableName, String columnName)
     }
 
     @Test
-    public void testMigrateUnsupportedColumnType()
+    public void testMigrateTimestampMicrosTypeWithAvro()
     {
-        String tableName = "test_migrate_unsupported_column_type_" + randomNameSuffix();
+        String tableName = "test_migrate_timestamp_micro_avro" + randomNameSuffix();
         String hiveTableName = "hive.tpch." + tableName;
-        String icebergTableName = "iceberg.tpch." + tableName;
+        Session session = timestampPrecisionSession(HiveTimestampPrecision.MICROSECONDS);
 
-        assertUpdate("CREATE TABLE " + hiveTableName + " AS SELECT timestamp '2021-01-01 00:00:00.000' x", 1);
+        assertUpdate(session, "CREATE TABLE " + hiveTableName + " (x timestamp(6)) WITH (format='AVRO')");
+        assertUpdate(session, "INSERT INTO " + hiveTableName + " VALUES timestamp '2021-01-01 00:00:00.123456'", 1);
+        // Currently Hive maps timestamp(6) to timestamp(millis)
+        assertQuery("SELECT * FROM " + hiveTableName, "VALUES timestamp '2021-01-01 00:00:00.123'");
 
-        assertQueryFails(
-                "CALL iceberg.system.migrate('tpch', '" + tableName + "')",
-                "\\QTimestamp precision (3) not supported for Iceberg. Use \"timestamp(6)\" instead.");
+        assertQueryFails("CALL iceberg.system.migrate('tpch', '" + tableName + "')", "Migrating timestamp type with AVRO format is not supported.");
+
+        assertUpdate("DROP TABLE " + hiveTableName);
+    }
 
+    @Test
+    public void testMigrateTimestampMillisTypeWithAvro()
+    {
+        String tableName = "test_migrate_timestamp_millis_avro" + randomNameSuffix();
+        String hiveTableName = "hive.tpch." + tableName;
+
+        assertUpdate("CREATE TABLE " + hiveTableName + " WITH (format='AVRO') AS SELECT timestamp '2021-01-01 00:00:00.000' x", 1);
         assertQuery("SELECT * FROM " + hiveTableName, "VALUES timestamp '2021-01-01 00:00:00.000'");
-        assertQueryFails("SELECT * FROM " + icebergTableName, "Not an Iceberg table: .*");
+
+        assertQueryFails("CALL iceberg.system.migrate('tpch', '" + tableName + "')", "Migrating timestamp type with AVRO format is not supported.");
 
         assertUpdate("DROP TABLE " + hiveTableName);
     }
 
+    public static Object[][] fileFormatTimestampPrecisionsAndValues()
```

Comment:
Can we run these tests with other session timezone as well? You can refer to `TestPostgreSqlTypeMapping#testTimestamp()`.

Reply:
As I replied in https://github.com/trinodb/trino/pull/17391#discussion_r1467238749, please avoid ParameterizedTest.

Reply:
done